### PR TITLE
Ports TG bamboo and Punji Sticks

### PR DESCRIPTION
**Pull request by chuga-git**
## What Does This PR Do

Ports https://github.com/tgstation/tgstation/pull/64027 (mostly)

- Adds bamboo as a growable plant, available in the seed vendor
    - Updates old bamboo harvest sprite
-  Adds craftable bamboo items and structures
    - bamboo and tatami mats
    - bamboo stools
    - bamboo benches 
    - ~~bamboo walls~~
- Adds craftable bamboo blow gun
    - Deals 20 stamina and 20 oxygen damage to the user per use
- Adds punji sticks, craftable using bamboo planks
    - Goes through shoes, inflicts a 6 second weaken when stepped on, which deals between 20 and 30 brute damage
    - Does not affect modsuit wearers (or other types of thick clothing)
- Adds an adjustable weaken_duration parameter to the caltrop component to control stun time (default is 6 SECONDS)

## Why It's Good For The Game

They will learn to fear the maintenance jungle. Also, fluff items are pretty cool.
## Images of changes

image

## Testing

(botany stuff)
- [x] Plant
    - [x] seed sprite
        - [x] Fix growth sprite negative z offset
    - [x] plant grows properly
    - [x] plant harvests properly
    - [x] plant sprites
    - [x] logs work properly
- [x] Logs and sheets merge and craft properly

(craftables)
- [x] Floor tiles
    - [x] Fix weird stack merging issues with tile stacks
    - [x] Bamboo mat placement and sprite rotation
    - [x] Tatami mats
    - [x] Tile smoothing
- [x] Wall and false wall sprites & smoothing
- [x] Benches
- [x] Stools (should be fixed)
- [x] Punji sticks
    - [x] Sprites and crafting
    - [x] Doesn't go through modsuits (or similar puncture immune clothing)
    - [x] Locker shenanigans (this can't be avoided without a full refactor of Crossed() as far as I can see)
    - [x] Placement shenanigans
- [x] Spear
- [x] Blowgun
- [x] Unfuck caltrop component because something has gone terribly wrong
## Changelog
:cl:
add: Bamboo: a growable plant available in the hydroponics seed vendor, along with craftable mats, stools, and benches.
add: Punji sticks: a deadly trap that can penetrate through shoes, craftable using bamboo.
add: Blowgun: a makeshift bamboo syringe gun that exhausts its user after too much use.
add: Bamboo spear: it's a spear made out of bamboo.
/:cl:

### CODE REVIEW

File: code/datums/components/caltrop.dm [hunk 1]
```diff
@@ -1,16 +1,29 @@
...
-	min_damage = _min_damage
-	max_damage = max(_min_damage, _max_damage)
-	probability = _probability
-	flags = _flags
+/datum/component/caltrop/Initialize(min_damage = 0, max_damage = 0, probability = 100, weaken_duration = 6 SECONDS, flags = NONE)
+	src.min_damage = min_damage
+	src.max_damage = max(min_damage, max_damage)
+	src.probability = probability
+	src.weaken_duration = weaken_duration
+	src.flags = flags
```
Comment (DGamerL) [inline code/datums/components/caltrop.dm:26]: Could maybe revert the changes to the argument removal of the `_`? Imo the previous version is more readable than this

File: code/modules/hydroponics/grown/towercap.dm [hunk 2]
```diff
@@ -102,6 +102,34 @@
...
+	AddComponent(/datum/component/caltrop, min_damage = 20, max_damage = 30, probability = 100, weaken_duration = 6 SECONDS, flags = CALTROP_BYPASS_SHOES)
+
+/obj/structure/punji_sticks/Crossed(mob/living/L)
+	return
+
+/obj/structure/punji_sticks/spikes
+	name = "wooden spikes"
+	icon_state = "woodspike"
+
+
```
Comment (DGamerL) [inline code/modules/hydroponics/grown/towercap.dm:132]:
```suggestion
```

File: code/modules/mining/lavaland/ash_flora.dm [hunk 3]
```diff
@@ -145,8 +145,7 @@
 
 /obj/structure/flora/ash/cacti/Initialize(mapload)
 	. = ..()
-	// min dmg 3, max dmg 6, prob(70)
-	AddComponent(/datum/component/caltrop, 3, 6, 70)
+	AddComponent(/datum/component/caltrop, min_damage = 3, max_damage = 6, probability = 70)
```
Comment (DGamerL) [inline code/modules/mining/lavaland/ash_flora.dm:149]:
```suggestion
```

**Comment by 1080pCat** [comment]
> Adds craftable bamboo blow gun
> 
>     * Deals 20 stamina and 20 oxygen damage per use
> 
>     * Details probably subject to change

good lord, craftable 20 stam gun? cant wait for people to craft 50 of them in a boh and rapid fire them...

File: code/game/objects/items/stacks/tiles/tile_types.dm [hunk 4]
```diff
@@ -68,6 +68,51 @@
...
+/obj/item/stack/tile/bamboo/tatami
+	name = "Tatami with green rim"
+	singular_name = "green tatami floor tile"
+	icon_state = "tile_tatami_green"
+	turf_type = /turf/simulated/floor/bamboo/tatami
+	merge_type = /obj/item/stack/tile/bamboo/tatami
+
+/obj/item/stack/tile/bamboo/tatami/twenty
+	amount = 20
+/obj/item/stack/tile/bamboo/tatami/purple
```
Comment (Henri215) [inline code/game/objects/items/stacks/tiles/tile_types.dm:96]:
```suggestion
	amount = 20

/obj/item/stack/tile/bamboo/tatami/purple
```

File: code/game/objects/items/stacks/sheets/sheet_types.dm [hunk 5]
```diff
@@ -265,6 +265,48 @@ GLOBAL_LIST_INIT(wood_recipes, list(
 	recipes = GLOB.wood_recipes
 	return ..()
 
+/*
+ * Bamboo
+ */
+GLOBAL_LIST_INIT(bamboo_recipes, list ( \
```
Comment (Henri215) [inline code/game/objects/items/stacks/sheets/sheet_types.dm:271]:
```suggestion
GLOBAL_LIST_INIT(bamboo_recipes, list( \
```

File: code/game/objects/items/stacks/sheets/sheet_types.dm [hunk 6]
```diff
@@ -265,6 +265,48 @@ GLOBAL_LIST_INIT(wood_recipes, list(
 	recipes = GLOB.wood_recipes
 	return ..()
 
+/*
+ * Bamboo
+ */
+GLOBAL_LIST_INIT(bamboo_recipes, list( \
+	new/datum/stack_recipe("punji sticks trap", /obj/structure/punji_sticks, req_amount = 5, time = 3 SECONDS, one_per_turf = TRUE, on_floor = TRUE), \
```
Comment (Henri215) [inline code/game/objects/items/stacks/sheets/sheet_types.dm:272]:
```suggestion
	new /datum/stack_recipe("punji sticks trap", /obj/structure/punji_sticks, req_amount = 5, time = 3 SECONDS, one_per_turf = TRUE, on_floor = TRUE),
```
Add a space there and remove the "\\" (not necessary) on the others.

File: code/game/objects/items/stacks/sheets/sheet_types.dm [hunk 7]
```diff
@@ -265,6 +265,48 @@ GLOBAL_LIST_INIT(wood_recipes, list(
...
+	new/datum/stack_recipe("bamboo spear", /obj/item/spear/bamboospear, req_amount = 25, time = 9 SECONDS), \
+	new/datum/stack_recipe("blow gun", /obj/item/gun/syringe/blowgun, req_amount = 10, time = 7 SECONDS), \
+	new/datum/stack_recipe("rice hat", /obj/item/clothing/head/rice_hat, req_amount = 10, time = 7 SECONDS), \
```
Comment (Henri215) [inline code/game/objects/items/stacks/sheets/sheet_types.dm:275]: it can now be made with bamboo and biogenerator?

File: code/game/objects/items/stacks/sheets/sheet_types.dm [hunk 8]
```diff
@@ -265,6 +265,48 @@ GLOBAL_LIST_INIT(wood_recipes, list(
...
+	null, \
+	new/datum/stack_recipe("bamboo stool", /obj/structure/chair/stool/bamboo, req_amount = 2, time = 1 SECONDS, on_floor = TRUE), \
+	new/datum/stack_recipe("bamboo mat piece", /obj/item/stack/tile/bamboo, req_amount = 1, res_amount = 4, max_res_amount = 20), \
+	new/datum/stack_recipe_list("tatami mats", list(
+		new/datum/stack_recipe("green tatami", /obj/item/stack/tile/bamboo/tatami, req_amount = 1, res_amount = 4, max_res_amount = 20),
+		new/datum/stack_recipe("purple tatami", /obj/item/stack/tile/bamboo/tatami/purple, req_amount = 1, res_amount = 4, max_res_amount = 20),
+		new/datum/stack_recipe("black tatami", /obj/item/stack/tile/bamboo/tatami/black, req_amount = 1, res_amount = 4, max_res_amount = 20),
+		)), \
+	null, \
+	new/datum/stack_recipe_list("bamboo benches", list(
```
Comment (Henri215) [inline code/game/objects/items/stacks/sheets/sheet_types.dm:285]: Remove the bamboo related recipes from the wood recipes.

File: code/game/objects/items/stacks/tiles/tile_types.dm [hunk 9]
```diff
@@ -68,6 +68,52 @@
...
+	singular_name = "purple tatami floor tile"
+	icon_state = "tile_tatami_purple"
+	turf_type = /turf/simulated/floor/bamboo/tatami/purple
+	merge_type = /obj/item/stack/tile/bamboo/tatami/purple
+
+/obj/item/stack/tile/bamboo/tatami/purple/twenty
+	amount = 20
+
+/obj/item/stack/tile/bamboo/tatami/black
+	name = "Tatami with black rim"
```
Comment (Henri215) [inline code/game/objects/items/stacks/tiles/tile_types.dm:108]:
```suggestion
	name = "tatami with purple rim"
	singular_name = "purple tatami floor tile"
	icon_state = "tile_tatami_purple"
	turf_type = /turf/simulated/floor/bamboo/tatami/purple
	merge_type = /obj/item/stack/tile/bamboo/tatami/purple

/obj/item/stack/tile/bamboo/tatami/purple/twenty
	amount = 20

/obj/item/stack/tile/bamboo/tatami/black
	name = "tatami with black rim"
```

File: code/game/objects/items/weapons/twohanded.dm [hunk 10]
```diff
@@ -320,6 +320,16 @@
 	throwforce = 22
 	armour_penetration_percentage = 15				//Enhanced armor piercing
 
+//Blatant imitation of spear, but all natural. Also not valid for explosive modification.
+/obj/item/spear/bamboospear
```
Comment (Henri215) [inline code/game/objects/items/weapons/twohanded.dm:324]:
```suggestion
/// Blatant imitation of spear, but all natural. Also not valid for explosive modification.
/obj/item/spear/bamboospear
```
Rename the path to just ```/obj/item/spear/bamboo```, no need to repeat the known fact that its a spear.

File: code/game/objects/items/weapons/twohanded.dm [hunk 11]
```diff
@@ -320,6 +320,16 @@
...
+	name = "bamboo spear"
+	desc = "A haphazardly-constructed bamboo stick with a sharpened tip, ready to poke holes into unsuspecting people."
+	base_icon_state = "bamboo_spear"
+	icon_state = "bamboo_spear0"
+	force = 10
+	force_unwielded = 10
+	force_wielded = 18
```
Comment (Henri215) [inline code/game/objects/items/weapons/twohanded.dm:331]:
```suggestion
```
Its inherited

File: code/modules/projectiles/guns/syringe_gun.dm [hunk 12]
```diff
@@ -399,3 +399,20 @@
...
+
+/// craftable bamboo syringe gun
+/obj/item/gun/syringe/blowgun
+	name = "blowgun"
+	desc = "Fire syringes at a short distance."
+	icon_state = "blowgun"
+	item_state = "gun"
+	fire_sound = 'sound/items/syringeproj.ogg'
+	w_class = WEIGHT_CLASS_NORMAL
+	force = 4
```
Comment (Henri215) [inline code/modules/projectiles/guns/syringe_gun.dm:411]:
```suggestion
```
Inherited

File: code/game/objects/items/stacks/tiles/tile_types.dm [hunk 13]
```diff
@@ -68,6 +68,52 @@
...
+	turf_type = /turf/simulated/floor/bamboo
+	merge_type = /obj/item/stack/tile/bamboo
+	parent_stack = TRUE
+	resistance_flags = FLAMMABLE
+
+/obj/item/stack/tile/bamboo/twenty
+	amount = 20
+
+/obj/item/stack/tile/bamboo/tatami
+	name = "Tatami with green rim"
```
Comment (Henri215) [inline code/game/objects/items/stacks/tiles/tile_types.dm:88]:
```suggestion
	name = "tatami with green rim"
```

**Comment by FunnyMan3595** [comment]
I've got a couple concerns here:
1. I'm not a fan of the sprites, and I've heard similar opinions from others. Please speak with our sprite team in Discord to figure out how to adjust them (and see if you can get help, if needed).
2. The blowgun and punji sticks, while neat, are adding yet more combat capabilities to Botany, which they really don't need right now.

**Review by Henri215 — APPROVED** [review]
Lewcs good to me

**Review by DGamerL — APPROVED** [review]

**Review by S34NW — APPROVED** [review]
